### ISSUE TITLE
Layer-wise precision in bf16, and fp8

### ISSUE THREAD

**Issue by bernardrb** [NVIDIA/TensorRT#3735]
## Description

We've read multiple issues that confirm the possibility of using bf16 (and fp8).  Currently, we think that these precisions can only be set using the trtexec or polygraph tool. While using the TensorRT Python API only supports setting precisions: int8, fp16, fp32 (not bf16, fp8).

We've seen the Nemo-demo, and are curious as to what would could grab from there, such that we can experiment with layer-precision. 

#3174 
#1883 
#3718

**TensorRT Version**: 9.3

**Comment by lix19937** [comment]
> FP8 : Enable FP8 layer selection     

from  8.6.1doc  https://docs.nvidia.com/deeplearning/tensorrt/api/python_api/infer/Core/BuilderConfig.html?highlight=builderflag%20sparse_weights    TensorRT Python API also supports `fp8 `, you can 
```py
config.set_flag(trt.BuilderFlag.FP8)    
```

from https://github.com/NVIDIA/TensorRT/blob/v9.1.0/include/NvInfer.h#L9144-L9152  v9.1.0 has already support fp8 and bf16, but the relevant documents have not been released from v9.1.0 .  

i guess the api like follow 
```
config.set_flag(trt.BuilderFlag.FP8)    
config.set_flag(trt.BuilderFlag.BF16)  
```

**Comment by lix19937** [comment]
More accurate reference you can see https://github.com/NVIDIA/TensorRT-LLM/blob/v0.8.0/tensorrt_llm/builder.py#L163-L169     

`tensorrt==9.2.0.post12.dev5`       
https://github.com/NVIDIA/TensorRT-LLM/blob/v0.8.0/requirements.txt#L15
